File: arrays_hashing/242_valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # time and space complexity of O(n+m)
        s_hash = {}
        t_hash = {}
        for thing in s:
            if thing in s_hash:
                s_hash[thing] += 1
            else:
                s_hash[thing] = 1
        for thing in t:
            if thing in t_hash:
                t_hash[thing] += 1
            else:
                t_hash[thing] = 1
        if t_hash == s_hash:
            return True
        else:
            return False

File: arrays_hashing/242_valid_anagram.cpp
#include <string>
#include <map>

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char, int> s_hash;
        map<char, int> t_hash;

        for ( char s_ch : s ) {
            s_hash[s_ch]++;
        }

        for ( char t_ch : t ) {
            t_hash[t_ch]++;
        }
        if (s_hash == t_hash) {
            return true;
        }
        else {
            return false;
        }
    }
};